File: backend/Services/Parsers/BoxOfficeGameParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class BoxOfficeGameParser : BasicScoreResultParser
    {
        private ILogger<BoxOfficeGameParser> _logger;

        public BoxOfficeGameParser(ILogger<BoxOfficeGameParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override string GameName => "Box Office Game";
        public override bool GolfScoring => false;
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"boxofficega\\.me.*🏆 (?<{ScoreGroup}>[\\d]+)", RegexOptions.Singleline);
        public override string Url => "https://boxofficega.me/";
        protected override string? ExtraContent => string.Empty;
        protected override string GetCleanResult(string result, Match parserResults)
        {
            return result.Replace("boxofficega.me", "boxofficegame");
        }
    }
}

File: lambdas/src/parsers/boxofficegame-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class BoxOfficeGameParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Box Office Game';
  override readonly golfScoring = false;
  readonly helpText = null;
  readonly url = 'https://boxofficega.me/';
  protected readonly extraContent = '';
  
  // Matches "boxofficega.me\n...\n🏆 920"
  protected readonly parser = new RegExp(
    `boxofficega\\.me.*?🏆 (?<${BasicScoreResultParser.SCORE_GROUP}>[\\d]+)`,
    's'
  );
  
  protected override getCleanResult(result: string, match: RegExpMatchArray): string {
    return result.replace('boxofficega.me', 'boxofficegame');
  }
}
